xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -7127,6 +7135,59 @@ ENTRY AddDotsFunc {
   EXPECT_TRUE(RunAndCompare(hlo_text, ErrorSpec{1e-5, 1e-5}));
 }
 
+// A test fixture class for tests which are specific to legacy cublas
+class SmallMatrixGemmRewriteTest : public GemmRewriteTest {
+ public:
+  DebugOptions GetDebugOptionsForTest() override {
+    DebugOptions debug_options = GemmRewriteTest::GetDebugOptionsForTest();
+    debug_options.set_xla_gpu_gemm_rewrite_size_threshold(100);
+    return debug_options;
+  }
+};
+
+TEST_F(SmallMatrixGemmRewriteTest, SkipSmallMatrixRewrite) {
+  const char* hlo_text = R"(
+HloModule SkipSmallMatrixRewrite
+
+ENTRY DotFunc {
+  x = f32[3,3] parameter(0)
+  y = f32[3,3] parameter(1)
+  ROOT out = f32[3,3] dot(x, y), lhs_contracting_dims={1}, rhs_contracting_dims={0}
+}
+)";
+
+  EXPECT_TRUE(RunAndCompare(hlo_text, ErrorSpec{1e-5, 1e-5}));
```

Comment:
nit: not sure if `RunAndCompare`s are necessary here (these should already be covered by other tests, either in the elemental rewriter or above in the same file for cuBLAS)